File: 3208. Alternating Groups II/main.go
package main

func numberOfAlternatingGroups(colors []int, k int) int {
	n := len(colors)
	res := 0
	l := 0

	for r := 1; r < n+k-1; r++ {
		if colors[r%n] == colors[(r-1)%n] {
			l = r
		}
		if r-l+1 == k {
			res++
			l++
		}
	}

	return res
}

File: 3208. Alternating Groups II/main.py
# total sam

from typing import List


class Solution:
    def numberOfAlternatingGroups(self, colors: List[int], k: int) -> int:
        n = len(colors)
        res = 0
        l = 0
        
        for r in range(1, n + k - 1):
            if colors[r%n] == colors[(r-1)%n]:
                l = r
            if r - l + 1 == k:
                res += 1
                l += 1

        return res